hooks/boot/02-message-helpers.rb

```diff
@@ -38,13 +38,24 @@ def dev_new_install_message(kafo)
   end
 
   def failure_message
-    say "\n  <%= color('There were errors detected during install.', :bad) %>"
-    say "  Please address the errors and re-run the installer to ensure the system is properly configured."
-    say "  Failing to do so is likely to result in broken functionality."
+    say "\n<%= color('There were errors detected during installation.', :bad) %>"
+
+    if puppet_report&.failed_resources&.any?
+      puppet_report.failed_resources.each do |failed_resource|
+        say "\nResource '#{failed_resource}' failed. Logs:"
```

Comment:
I think we could drop `Resource` here since the Kafo output already includes that:

Before:
```
Resource Puppet Exec resource 'foreman-rake-db:migrate' failed
```

After
```
Puppet Exec resource 'foreman-rake-db:migrate' failed
```